spec/support/helpers/state_file_intake_helper.rb

```diff
@@ -133,6 +141,19 @@ def step_through_df_data_transfer(sample_name = "Transfer my #{filing_year} fede
     click_on I18n.t("general.continue") if expect_success
   end
 
+  def page_change_check(text, sleep_time: 0.1)
+    begin
+      expect(page).to have_text(text)
+    rescue Selenium::WebDriver::Error::WebDriverError,
+      Capybara::ElementNotFound,
+      RSpec::Expectations::ExpectationNotMetError => e
+      puts "Caught #{e.class} - #{e.message}"
+      puts "First attempt failed for `#{text}`, sleeping #{sleep_time} seconds then retrying..."
+      sleep sleep_time
+      retry
+    end
+  end
```

Comment:
do we want to implement a max number of times to attempt, otherwise we maybe stuck forever retrying

Reply:
Oh interesting! I didn't know thats how retry worked. Yes In will try changing to 2.times { retry }

Reply:
I think you'll want to keep a counter in this method, not sure if that syntax works

Reply:
yes lol i tried that and was like why is that failing 😂 the counter works, added a retry param too